lib/matplotlib/axes/_axes.py

```diff
@@ -6666,19 +6681,19 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
         # If bins are not specified either explicitly or via range,
         # we need to figure out the range required for all datasets,
         # and supply that to np.histogram.
-        if not binsgiven and not input_empty:
-            xmin = np.inf
-            xmax = -np.inf
-            for xi in x:
-                if len(xi) > 0:
-                    xmin = min(xmin, np.nanmin(xi))
-                    xmax = max(xmax, np.nanmax(xi))
-            bin_range = (xmin, xmax)
```

Comment:
You either need to keep this code for the case when `bins` is an integer, or you need to add a case for integral bins in the `histogram_bin_edges` backport

Reply:
ah, good catch on that, I fell down a different rabbit hole (see my comment at the issue level)...